### PR TITLE
add borrower tests - lock

### PR DESCRIPTION
**Pull request by ilinzweilin**
- added system tests for nft locks
- refactor: added custom setup functions to tests
- made SystemTest DSTest and removed DSTest from TestSetup

### CODE REVIEW

File: src/core/test/system/borrower/close.t.sol [hunk 1]
```diff
@@ -50,7 +59,8 @@ contract CloseTest is SystemTest {
         uint loanId = borrower.issue(collateralNFT_, tokenId);
         // transfer nft to random user / borrower still loanOwner
         borrower.approveNFT(collateralNFT, address(this));
-        collateralNFT.transferFrom(borrower_, address(this), tokenId);  
+        collateralNFT.transferFrom(borrower_, address(this), tokenId);
```
Comment (manuelpolzhofer) [inline src/core/test/system/borrower/close.t.sol]: It could be any address here correct? you just want to ensure the loan owner isn't owning the collateral nft.

File: src/core/test/system/borrower/lock.sol [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        lockNFT(loanId, tokenId);
+    }
+
+    function testFailLockNFTBorrowerNotNFTOwner() public {
+        (uint tokenId, ) = issueNFT(borrower_);
+        // borrower creates loan against nft
+        uint loanId = borrower.issue(collateralNFT_, tokenId);
+        // borrower transfers nftOwnership to random user / borrower still loanOwner
+        borrower.approveNFT(collateralNFT, address(this));
+        collateralNFT.transferFrom(borrower_, address(this), tokenId);
```
Comment (manuelpolzhofer) [inline src/core/test/system/borrower/lock.sol]: maybe not use `address(this)` if it should be random
```suggestion
        collateralNFT.transferFrom(borrower_, address(123), tokenId); 
```

File: src/core/test/system/borrower/lock.sol [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        // borrower transfers nftOwnership to random user / borrower still loanOwner
+        borrower.approveNFT(collateralNFT, address(this));
+        collateralNFT.transferFrom(borrower_, address(this), tokenId); 
+        // nftOwner approves shelf to lock
+        collateralNFT.setApprovalForAll(address(shelf), true);
+        // borrower tries to lock nft
+        lockNFT(loanId, tokenId);
+    }
+
+    function testFailLockNFTBorrowerNotLoanOwner() public {
```
Comment (manuelpolzhofer) [inline src/core/test/system/borrower/lock.sol]: good edge cases 👍

File: src/core/test/system/lender/supply.t.sol [hunk 4]
```diff
@@ -19,6 +19,10 @@ import "../system.t.sol";
 
 contract SupplyTest is SystemTest {
 
+    function setUp() public {
+        baseSetup();
```
Comment (manuelpolzhofer) [inline src/core/test/system/lender/supply.t.sol]: to keep the pattern should we not create the investor user here?

File: src/core/test/system/system.t.sol [hunk 5]
```diff
@@ -16,9 +16,24 @@
 pragma solidity >=0.5.12;
 
 import "./system.sol";
+import "./users/borrower.sol";
 
 contract STest is SystemTest {
```
Comment (manuelpolzhofer) [inline src/core/test/system/system.t.sol]: What is STest just checking if systemTest works?

File: src/core/test/system/lender/supply.t.sol (hunk 4 above)
Comment (ilinzweilin) [reply src/core/test/system/lender/supply.t.sol]: yes, already talked with @charleenfei

File: src/core/test/system/system.t.sol (hunk 5 above)
Comment (ilinzweilin) [reply src/core/test/system/system.t.sol]: STest are our old system tests, we will get rid of this file eventually

File: src/core/test/system/borrower/lock.sol (hunk 2 above)
Comment (ilinzweilin) [reply src/core/test/system/borrower/lock.sol]: yes, makes sense for better readability, will change for all the tests
